File: mpp_proiect_java/TriatlonNetworkingProtoBuff/src/main/java/triatlon/network/protobuffprotocol/ProtoUtils.java
package triatlon.network.protobuffprotocol;

import com.google.protobuf.Descriptors;
import domain.*;

import java.util.ArrayList;
import java.util.List;

public class ProtoUtils {
    public static TriatlonProtobufs.TriatlonRequest createLoginRequest(Arbitru user){
        TriatlonProtobufs.Arbitru userDTO=TriatlonProtobufs.Arbitru.newBuilder().setId(user.getId()).setFirstname(user.getFirstName()).setLastname(user.getLastName()).setEmail(user.getEmail()).setUserName(user.getUserName()).setPassword(user.getPassword()).setResponsabilProba(user.getIdProbaArbitru()).build();
        TriatlonProtobufs.TriatlonRequest request= TriatlonProtobufs.TriatlonRequest.newBuilder().setType(TriatlonProtobufs.TriatlonRequest.Type.Login)
                .setArbitru(userDTO).build();
        return request;
    }
    public static TriatlonProtobufs.TriatlonRequest createLogoutRequest(Arbitru user){
        TriatlonProtobufs.Arbitru userDTO=TriatlonProtobufs.Arbitru.newBuilder().setId(user.getId()).setFirstname(user.getFirstName()).setLastname(user.getLastName()).setEmail(user.getEmail()).setUserName(user.getUserName()).setPassword(user.getPassword()).setResponsabilProba(user.getIdProbaArbitru()).build();
        TriatlonProtobufs.TriatlonRequest request= TriatlonProtobufs.TriatlonRequest.newBuilder().setType(TriatlonProtobufs.TriatlonRequest.Type.Logout)
                .setArbitru(userDTO).build();
        return request;
    }

    public static TriatlonProtobufs.TriatlonRequest creategetArbitrubyNameRequest(String nume){
        TriatlonProtobufs.StringSS stringSS = TriatlonProtobufs.StringSS.newBuilder().setNameArbitru(nume).build();
        TriatlonProtobufs.TriatlonRequest request= TriatlonProtobufs.TriatlonRequest.newBuilder()
                .setType(TriatlonProtobufs.TriatlonRequest.Type.getArbitrubyName).setNameArbitru(stringSS).build();
        return request;
    }
    public static TriatlonProtobufs.TriatlonRequest createfindParticipantByNumePrenumeRequest(String nume){
        TriatlonProtobufs.StringSS stringSS = TriatlonProtobufs.StringSS.newBuilder().setNameArbitru(nume).build();
        TriatlonProtobufs.TriatlonRequest request= TriatlonProtobufs.TriatlonRequest.newBuilder()
                .setType(TriatlonProtobufs.TriatlonRequest.Type.findParticipantByNumePrenume).setNameArbitru(stringSS).build();
        return request;
    }

    public static TriatlonProtobufs.TriatlonRequest creategetProbaArbitrubyIdRequest(long id){
        TriatlonProtobufs.LonggSS longSS = TriatlonProtobufs.LonggSS.newBuilder().setIdEntity(id).build();
        TriatlonProtobufs.TriatlonRequest request= TriatlonProtobufs.TriatlonRequest.newBuilder()
                .setType(TriatlonProtobufs.TriatlonRequest.Type.getProbaArbitrubyId).setIdEntity(longSS).build();
        return request;
    }
    public static TriatlonProtobufs.TriatlonRequest createGetParticipantiDTORequest(Arbitru user){
        TriatlonProtobufs.Arbitru userDTO=TriatlonProtobufs.Arbitru.newBuilder().setId(user.getId()).setFirstname(user.getFirstName()).setLastname(user.getLastName()).setEmail(user.getEmail()).setUserName(user.getUserName()).setPassword(user.getPassword()).setResponsabilProba(user.getIdProbaArbitru()).build();
        TriatlonProtobufs.TriatlonRequest request= TriatlonProtobufs.TriatlonRequest.newBuilder().setType(TriatlonProtobufs.TriatlonRequest.Type.getParticipantiDTO)
                .setArbitru(userDTO).build();
        return request;
    }
    public static TriatlonProtobufs.TriatlonRequest createfilterbyProbaRequest(Proba proba){
        TriatlonProtobufs.Proba probaPr=TriatlonProtobufs.Proba.newBuilder().setId(proba.getId()).setTipproba(proba.getTipProba()).setDistanta(proba.getDistanta()).build();
        TriatlonProtobufs.TriatlonRequest request= TriatlonProtobufs.TriatlonRequest.newBuilder().setType(TriatlonProtobufs.TriatlonRequest.Type.filterbyProba)
                .setProba(probaPr).build();
        return request;
    }

    public static TriatlonProtobufs.TriatlonRequest createSendRezultatRequest(Rezultat rezultat){
        TriatlonProtobufs.Proba probaPr = TriatlonProtobufs.Proba.newBuilder().setId(rezultat.getProba().getId()).setTipproba(rezultat.getProba().getTipProba()).setDistanta(rezultat.getProba().getDistanta()).build();
        TriatlonProtobufs.Participant participantPr = TriatlonProtobufs.Participant.newBuilder().setId(rezultat.getParticipant().getId().toString()).setFirstname(rezultat.getParticipant().getFirstName()).setLastname(rezultat.getParticipant().getLastName()).setVarsta(rezultat.getParticipant().getVarsta().toString()).build();

        TriatlonProtobufs.Rezultat rezultatR=TriatlonProtobufs.Rezultat.newBuilder().setProba(probaPr).setParticipant(participantPr).setNumarpuncte(rezultat.getNumarPuncte()).build();
        TriatlonProtobufs.TriatlonRequest request= TriatlonProtobufs.TriatlonRequest.newBuilder().setType(TriatlonProtobufs.TriatlonRequest.Type.SendRezultat)
                .setRezultat(rezultatR).build();
        return request;
    }
//    public static TriatlonProtobufs.TriatlonRequest createSendRezultatRequest(Rezultat rezultat){
//        TriatlonProtobufs.Rezultat messageDTO=TriatlonProtobufs.Rezultat.newBuilder().
//                setId(rezultat.getId())
//                .setParticipant(rezultat.getParticipant())
//                .setProba(rezultat.getProba())
//                .setNumarpuncte(rezultat.getNumarPuncte())
//                .build();
//
//
//        TriatlonProtobufs.ChatRequest request= ChatProtobufs.ChatRequest.newBuilder()
//                .setType(ChatProtobufs.ChatRequest.Type.SendMessage)
//                .setMessage(messageDTO).build();
//        return request;
//    }
//
//    public static ChatProtobufs.ChatRequest createLoggedFriendsRequest(User user){
//        ChatProtobufs.User userDTO=ChatProtobufs.User.newBuilder().setId(user.getId()).build();
//        ChatProtobufs.ChatRequest request= ChatProtobufs.ChatRequest.newBuilder()
//                .setType(ChatProtobufs.ChatRequest.Type.GetLoggedFriends)
//                .setUser(userDTO).build();
//        return request;
//    }
//
//
//    public static ChatProtobufs.ChatResponse createOkResponse(){
//        ChatProtobufs.ChatResponse response=ChatProtobufs.ChatResponse.newBuilder()
//                .setType(ChatProtobufs.ChatResponse.Type.Ok).build();
//        return response;
//    }
//
//    public static ChatProtobufs.ChatResponse createErrorResponse(String text){
//        ChatProtobufs.ChatResponse response=ChatProtobufs.ChatResponse.newBuilder()
//                .setType(ChatProtobufs.ChatResponse.Type.Error)
//                .setError(text).build();
//        return response;
//    }
//
//    public static ChatProtobufs.ChatResponse createFriendLoggedInResponse(User user){
//        ChatProtobufs.User userDTO=ChatProtobufs.User.newBuilder().setId(user.getId()).build();
//
//        ChatProtobufs.ChatResponse response=ChatProtobufs.ChatResponse.newBuilder()
//                .setType(ChatProtobufs.ChatResponse.Type.FriendLoggedIn)
//                .setUser(userDTO).build();
//        return response;
//    }
//
//    public static ChatProtobufs.ChatResponse createFriendLoggedOutResponse(User user){
//        ChatProtobufs.User userDTO=ChatProtobufs.User.newBuilder().setId(user.getId()).build();
//
//        ChatProtobufs.ChatResponse response=ChatProtobufs.ChatResponse.newBuilder()
//                .setType(ChatProtobufs.ChatResponse.Type.FriendLoggedOut)
//                .setUser(userDTO).build();
//        return response;
//    }
//    public static TriatlonProtobufs.TriatlonResponse createNewMessageResponse(Message message){
//        ChatProtobufs.Message messageDTO=ChatProtobufs.Message.newBuilder()
//                .setSenderId(message.getSender().getId())
//                .setReceiverId(message.getReceiver().getId())
//                .setText(message.getText())
//                .build();
//
//        ChatProtobufs.ChatResponse response=ChatProtobufs.ChatResponse.newBuilder()
//                .setType(ChatProtobufs.ChatResponse.Type.NewMessage)
//                .setMessage(messageDTO).build();
//        return response;
//    }

//    public static ChatProtobufs.ChatResponse createLoggedFriendsResponse(User[] users){
//        ChatProtobufs.ChatResponse.Builder response=ChatProtobufs.ChatResponse.newBuilder()
//                .setType(ChatProtobufs.ChatResponse.Type.GetLoggedFriends);
//        for (User user: users){
//            ChatProtobufs.User userDTO=ChatProtobufs.User.newBuilder().setId(user.getId()).build();
//            response.addFriends(userDTO);
//        }
//
//        return response.build();
//    }
//
    public static String getError(TriatlonProtobufs.TriatlonResponse response){
        String errorMessage=response.getError();
        return errorMessage;
    }


//    public static Message getMessage(ChatProtobufs.ChatResponse response){
//        User sender=new User();
//        sender.setId(response.getMessage().getSenderId());
//        User receiver=new User();
//        receiver.setId(response.getMessage().getReceiverId());
//        Message message=new Message(sender,response.getMessage().getText(), receiver);
//        return message;
//    }
//
//    public static User[] getFriends(ChatProtobufs.ChatResponse response){
//        User[] friends=new User[response.getFriendsCount()];
//        for(int i=0;i<response.getFriendsCount();i++){
//            ChatProtobufs.User userDTO=response.getFriends(i);
//            User user=new User();
//            user.setId(userDTO.getId());
//            friends[i]=user;
//        }
//        return friends;
//    }
    public static Arbitru getArbitru(TriatlonProtobufs.TriatlonResponse response){
        Arbitru user=new Arbitru(response.getArbitru().getId(),response.getArbitru().getFirstname(),response.getArbitru().getLastname(),response.getArbitru().getEmail(), response.getArbitru().getPassword(),response.getArbitru().getUserName(), response.getArbitru().getResponsabilProba());
        return user;
    }
    public static Participant getParticipant(TriatlonProtobufs.TriatlonResponse response){
        Participant part = new Participant(Long.parseLong(response.getParticipant().getId()),response.getParticipant().getFirstname(),response.getParticipant().getLastname(),Integer.parseInt(response.getParticipant().getVarsta()));
        return part;
    }


    public static Proba getProba(TriatlonProtobufs.TriatlonResponse response){
        Proba proba = new Proba(response.getProba().getId(),response.getProba().getTipproba(),response.getProba().getDistanta());
        return proba;
    }
    public static Rezultat getRezultat(TriatlonProtobufs.TriatlonResponse response){
        Proba proba = new Proba(response.getRezultat().getProba().getId(),response.getRezultat().getProba().getTipproba(),response.getRezultat().getProba().getDistanta());
        Participant participant = new Participant(Long.parseLong(response.getRezultat().getParticipant().getId()),response.getRezultat().getParticipant().getFirstname(),response.getRezultat().getParticipant().getLastname(),Integer.parseInt(response.getRezultat().getParticipant().getVarsta()));
        Rezultat rez = new Rezultat(response.getRezultat().getId(),proba,participant,response.getRezultat().getNumarpuncte());
        return rez;
    }
    public static List<ParticipantDTO> getParticipantiDTO(TriatlonProtobufs.TriatlonResponse response){
        List<ParticipantDTO> listaP = new ArrayList<>();

        for(int i=0;i<response.getParticipantiiDTOCount();i++){
            TriatlonProtobufs.ParticipantDTO participantDTO=response.getParticipantiiDTO(i);
            ParticipantDTO participant=new ParticipantDTO(participantDTO.getFirstname(),participantDTO.getLastname(), participantDTO.getPunctaj());
            listaP.add(participant);
        }
        return listaP;
    }
    public static List<Rezultat> getfilterbyProbaRequest(TriatlonProtobufs.TriatlonResponse response){
        List<Rezultat> listaR = new ArrayList<>();
        for(int i=0;i<response.getRezultateCount();i++){
            TriatlonProtobufs.Rezultat rezultat=response.getRezultate(i);
            Proba proba = new Proba(rezultat.getId(),rezultat.getProba().getTipproba(),rezultat.getProba().getDistanta());
            Participant participant = new Participant(Long.parseLong(rezultat.getParticipant().getId()),rezultat.getParticipant().getFirstname(),rezultat.getParticipant().getLastname(), Integer.parseInt(rezultat.getParticipant().getVarsta()));
            Rezultat rez=new Rezultat(rezultat.getId(),proba,participant,rezultat.getNumarpuncte());
            listaR.add(rez);
        }
        return listaR;
    }
//
//    public static Participant getParticipant(TriatlonProtobufs.TriatlonRequest request){
//        Participant user=new Participant(Long. parseLong(request.getParticipant().getId()),request.getParticipant().getFirstname(),request.getParticipant().getLastname(), Integer. valueOf(request.getParticipant().getVarsta()));
//
//        return user;
//    }
////
//    public static Message getMessage(ChatProtobufs.ChatRequest request){
//        User sender=new User();
//        sender.setId(request.getMessage().getSenderId());
//        User receiver=new User();
//        receiver.setId(request.getMessage().getReceiverId());
//        Message message=new Message(sender,request.getMessage().getText(), receiver);
//        return message;
//    }

}

File: mpp_proiect_CSharp/triatlon/protobuff3/ProtoUtils.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Triatlon.Protocol;
using proto = Triatlon.Protocol;
namespace protobuf
{
    static class ProtoUtils
    {
        public static TriatlonRequest createLoginRequest(triatlon.domain.Arbitru user)
        {
            //new proto.User {Id = user.Id, Passwd = user.Password};
            proto.Arbitru userDTO = new proto.Arbitru { UserName = user.username, Password = user.password };
            TriatlonRequest request = new TriatlonRequest { Type = TriatlonRequest.Types.Type.Login, Arbitru = userDTO };

            return request;
        }

        public static TriatlonRequest createLogoutRequest(triatlon.domain.Arbitru user)
        {
            proto.Arbitru userDTO = new proto.Arbitru { UserName = user.username, Password = user.password };
            TriatlonRequest request = new TriatlonRequest { Type = TriatlonRequest.Types.Type.Logout, Arbitru = userDTO };

            return request;
        }

        public static TriatlonRequest createGetParticipantiiDTORequest(triatlon.domain.Arbitru user)
        {
            proto.Arbitru userDTO = new proto.Arbitru { UserName = user.username,Email=user.email,Firstname = user.firstName,Lastname = user.lastName, Password = user.password };
            TriatlonRequest request = new TriatlonRequest { Type = TriatlonRequest.Types.Type.GetParticipantiDto, Arbitru = userDTO };

            return request;
        }



        //public static ChatRequest createSendMesssageRequest(chat.model.Message message)
        //{
        //    proto.Message messageDTO = new proto.Message
        //    {
        //        SenderId = message.Sender.Id,
        //        ReceiverId = message.Receiver.Id,
        //        Text = message.Text
        //    };

        //    ChatRequest request = new ChatRequest { Type = ChatRequest.Types.Type.SendMessage, Message = messageDTO };
        //    return request;
        //}




        public static TriatlonResponse createOkResponse()
        {
            TriatlonResponse response = new TriatlonResponse { Type = TriatlonResponse.Types.Type.Ok };
            return response;
        }


        public static TriatlonResponse createErrorResponse(String text)
        {
            TriatlonResponse response = new TriatlonResponse
            {
                Type = TriatlonResponse.Types.Type.Error,
                Error = text
            };
            return response;
        }

        //public static ChatResponse createFriendLoggedInResponse(chat.model.User user)
        //{
        //    proto.User userDTO = new proto.User { Id = user.Id };
        //    ChatResponse response = new ChatResponse { Type = ChatResponse.Types.Type.FriendLoggedIn, User = userDTO };
        //    return response;
        //}

        //public static ChatResponse createFriendLoggedOutResponse(chat.model.User user)
        //{
        //    proto.User userDTO = new proto.User { Id = user.Id };
        //    ChatResponse response = new ChatResponse { Type = ChatResponse.Types.Type.FriendLoggedOut, User = userDTO };
        //    return response;
        //}

        //public static ChatResponse createNewMessageResponse(chat.model.Message message)
        //{
        //    proto.Message messageDTO = new proto.Message
        //    {
        //        SenderId = message.Sender.Id,
        //        ReceiverId = message.Receiver.Id,
        //        Text = message.Text
        //    };

        //    ChatResponse response = new ChatResponse { Type = ChatResponse.Types.Type.NewMessage, Message = messageDTO };
        //    return response;
        //}


        public static TriatlonResponse  createGetArbitruByNameResponse(triatlon.domain.Arbitru arbitru)
        {
           
            proto.Arbitru arbitruResponse = new proto.Arbitru
            {
                Id = Convert.ToUInt64(arbitru.Id),
                Firstname = arbitru.firstName,
                Lastname = arbitru.lastName,
                Email = arbitru.email,
                Password = arbitru.password,
                ResponsabilProba = Convert.ToUInt64(arbitru.id_proba_responsabil),
                UserName = arbitru.username
            };

            TriatlonResponse response = new TriatlonResponse { Type = TriatlonResponse.Types.Type.GetArbitrubyName, Arbitru = arbitruResponse };
            return response;
        }
        public static TriatlonResponse createFindParticipantByNumePrenumeResponse(triatlon.domain.Participant part)
        {
            proto.Participant participant = new proto.Participant
            {
                Id =part.Id.ToString(),
                Firstname = part.firstName,
                Lastname = part.lastName,
                Varsta = part.varsta.ToString()
               
            };

            TriatlonResponse response = new TriatlonResponse { Type = TriatlonResponse.Types.Type.FindParticipantByNumePrenume, Participant = participant };
            return response;
        }
        public static TriatlonResponse createProbaArbitrubyIdResponse(triatlon.domain.Proba proba)
        {
            proto.Proba probaResponse = new proto.Proba
            {
                Id = Convert.ToUInt64(proba.Id),
                Distanta = Convert.ToUInt32(proba.distanta),
                Tipproba = proba.tipProba
               
            };
            TriatlonResponse response = new TriatlonResponse { Type = TriatlonResponse.Types.Type.GetProbaArbitrubyId, Proba = probaResponse };
            return response;
        }

        public static TriatlonResponse createGetParticipantiiGtoResponse(triatlon.domain.ParticipantDTO[] participantii)
        {
            TriatlonResponse response = new TriatlonResponse
            {
                Type = TriatlonResponse.Types.Type.GetParticipantiDto
            };
            foreach (triatlon.domain.ParticipantDTO user in participantii)
            {
                proto.ParticipantDTO userDTO = new proto.ParticipantDTO { Firstname = user.firstName,Lastname = user.lastName,Punctaj = Convert.ToUInt32(user.punctaj) };

                response.ParticipantiiDTO.Add(userDTO);
            }

            return response;
        }
        public static TriatlonResponse createFilterProbaResponse(triatlon.domain.Rezultat[] rezultate)
        {
            TriatlonResponse response = new TriatlonResponse
            {
                Type = TriatlonResponse.Types.Type.FilterbyProba
            };
            foreach (triatlon.domain.Rezultat rez in rezultate)
            {
                proto.Participant participantX = new proto.Participant {Id = rez.participant.Id.ToString(),Firstname = rez.participant.firstName,Lastname = rez.participant.lastName,Varsta = rez.participant.varsta.ToString() };
                proto.Proba probaX = new proto.Proba { Id = Convert.ToUInt64(rez.proba.Id),Distanta = Convert.ToUInt32(rez.proba.distanta),Tipproba = rez.proba.tipProba};
                proto.Rezultat rezultat = new proto.Rezultat { Id = Convert.ToUInt64(rez.Id), Numarpuncte = Convert.ToUInt32(rez.numarPuncte), Participant =participantX,Proba = probaX};

                response.Rezultate.Add(rezultat);
            }

            return response;
        }

            public static String getError(TriatlonResponse response)
        {
            String errorMessage = response.Error;
            return errorMessage;
        }

        public static triatlon.domain.Arbitru getArbitru(TriatlonRequest response)
        {
            triatlon.domain.Arbitru user = new triatlon.domain.Arbitru(response.Arbitru.UserName,response.Arbitru.Password);
            return user;
        }
        public static triatlon.domain.Proba getProba(TriatlonRequest response)
        {
            triatlon.domain.Proba proba = new triatlon.domain.Proba(Convert.ToInt64(response.Proba.Id), response.Proba.Tipproba, Convert.ToInt32(response.Proba.Distanta));
            return proba;
        }
        public static triatlon.domain.Rezultat getRezultat(TriatlonRequest response)
        {
            triatlon.domain.Proba proba = new triatlon.domain.Proba(Convert.ToInt64(response.Rezultat.Proba.Id), response.Rezultat.Proba.Tipproba, Convert.ToInt32(response.Rezultat.Proba.Distanta));
            triatlon.domain.Participant participant = new triatlon.domain.Participant(Convert.ToInt64(response.Rezultat.Participant.Id),response.Rezultat.Participant.Firstname,response.Rezultat.Participant.Lastname,Convert.ToInt32(response.Rezultat.Participant.Varsta));
            triatlon.domain.Rezultat rez = new triatlon.domain.Rezultat(Convert.ToInt64(response.Rezultat.Id),proba,participant,Convert.ToInt32(response.Rezultat.Numarpuncte));
            return rez;
        }
        public static TriatlonResponse createNewRezultatResponse(triatlon.domain.Rezultat rez)
        {

            proto.Participant participantX = new proto.Participant { Id = rez.participant.Id.ToString(), Firstname = rez.participant.firstName, Lastname = rez.participant.lastName, Varsta = rez.participant.varsta.ToString() };
            proto.Proba probaX = new proto.Proba { Id = Convert.ToUInt64(rez.proba.Id), Distanta = Convert.ToUInt32(rez.proba.distanta), Tipproba = rez.proba.tipProba };
            proto.Rezultat rezultat = new proto.Rezultat { Id = Convert.ToUInt64(rez.Id), Numarpuncte = Convert.ToUInt32(rez.numarPuncte), Participant = participantX, Proba = probaX };


            TriatlonResponse response = new TriatlonResponse { Type = TriatlonResponse.Types.Type.NewRezultat, Rezultat = rezultat };
            return response;
        }

        public static string getNumeArbitru(TriatlonRequest response)
        {
            string nume_arbitru = response.NameArbitru.NameArbitru;
            return nume_arbitru;
        }
        public static long getIdArbitru(TriatlonRequest response)
        {
            long id_arb = Convert.ToInt64(response.IdEntity.IdEntity);
            return id_arb;
        }
        


        //public static chat.model.Message getMessage(ChatResponse response)
        //{
        //    chat.model.User sender = new chat.model.User(response.Message.SenderId);
        //    chat.model.User receiver = new chat.model.User(response.Message.ReceiverId);
        //    chat.model.Message message = new chat.model.Message(sender, receiver, response.Message.Text);
        //    return message;
        //}

        //public static chat.model.User[] getFriends(ChatResponse response)
        //{
        //    chat.model.User[] friends = new chat.model.User[response.Friends.Count];
        //    for (int i = 0; i < response.Friends.Count; i++)
        //    {
        //        chat.model.User user = new chat.model.User(response.Friends[i].Id);
        //        friends[i] = user;
        //    }
        //    return friends;
        //}

        //public static chat.model.User getUser(ChatRequest request)
        //{
        //    chat.model.User user = new chat.model.User(request.User.Id);
        //    user.Password = request.User.Passwd;
        //    return user;
        //}
        //public static chat.model.Message getMessage(ChatRequest request)
        //{
        //    chat.model.User sender = new chat.model.User(request.Message.SenderId);
        //    chat.model.User receiver = new chat.model.User(request.Message.ReceiverId);
        //    chat.model.Message message = new chat.model.Message(sender, receiver, request.Message.Text);
        //    return message;
        //}

    }
}